src/app-shared/core/service/web-frame-context/web-frame-context-state.service.ts

```diff
@@ -4,19 +4,51 @@
  * Available under MIT license webFrame/LICENSE
  */
 
-import { Injectable, InjectionToken } from '@angular/core';
+import { Inject, Injectable, InjectionToken } from '@angular/core';
 import { User } from 'app-shared/core';
-import { BehaviorSubject } from 'rxjs';
+import { BehaviorSubject, Observable } from 'rxjs';
+import { map } from 'rxjs/operators';
+import { ServerApi } from '../api-endpoint-builder/api-endpoint-builder.interface';
+import { IWebRequestService, RequestConfiguration } from '../web-request/web-request.interface';
 
 /**
  * Service holding the state of the application.
  */
 export interface IWebFrameContextStateService {
+  ready: boolean;
   currentUser: BehaviorSubject<User>;
+
+  initialize(): Observable<boolean>;
 }
 export const IWebFrameContextStateService = new InjectionToken('IWebFrameContextStateService');
-
+// TODO: Move outside
 @Injectable()
 export class WebFrameContextStateService implements IWebFrameContextStateService {
   public currentUser: BehaviorSubject<User>;
+  public ready: boolean;
```

Comment:
I think that you should avoid using this flag. The BehaviorSubject value can be accessed via it's `.value` property. And you should define it like:
```
public readonly currentUser: BehaviorSubject<User> = new BehaviorSubject<User>(null);
```